### PR TITLE
Manager longevities and corrupt then manager repair nemesis

### PR DESCRIPTION
**Pull request by ShlomiBalalis**
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by ShlomiBalalis** [comment]
@roydahan Those are just a few options that I thought could be beneficial for testing. Also considered adding an operator scenario, but wasn't sure since I see we are not testing those on releases

File: jenkins-pipelines/manager/longevity-large-partition-asymmetric-cluster-ManagerOperationsMonkey-3h.jenkinsfile
Comment (roydahan) [inline jenkins-pipelines/manager/longevity-large-partition-asymmetric-cluster-ManagerOperationsMonkey-3h.jenkinsfile:1]: If you picked the 8h I think this one isn't needed

**Review by roydahan — CHANGES_REQUESTED** [review]
All looks good, just remove the unnecessary new class and use nemesis_selector.

File: jenkins-pipelines/manager/longevity-large-partition-asymmetric-cluster-ManagerOperationsMonkey-3h.jenkinsfile
Comment (ShlomiBalalis) [reply jenkins-pipelines/manager/longevity-large-partition-asymmetric-cluster-ManagerOperationsMonkey-3h.jenkinsfile:1]: A line in the release notes made me think this test will be worth it:
`A Repair job always starts on the host with the smallest shards (cores) count`

Comment (ShlomiBalalis) [reply jenkins-pipelines/manager/longevity-large-partition-asymmetric-cluster-ManagerOperationsMonkey-3h.jenkinsfile:1]: And I've yet to find an option to do so in dtest

**Comment by Michal-Leszczynski** [comment]
I don't have enough SCT knowledge to validate those changes, but here are some thoughts in terms of testing repair (perhaps it's going to be useful in some other tests).

If you repair all nodes in the cluster, you can expect that SM should schedule repair jobs for all token ranges of repaired table. This means, that scheduled token ranges fully cover Int64 range [min_int_64, max_int_64]. But perhaps it's too implementation specific to include in a test?

Here is some other scenario:
- repair task starts
- it encounters errors on some ranges but it proceeds (`--fail-fast=false`)
- repair task is paused
- repair is resumed - SM will try to repair failed ranges again

Maybe it would be nice to test that SM retries failed ranges on pause/resume? But I'm not sure how you can force Scylla to return error for given repair job (in SM repo we do that by setting interceptors on client used to communicate with Scylla).

File: sdcm/nemesis.py [hunk 1]
```diff
@@ -6284,3 +6314,11 @@ class DisableBinaryGossipExecuteMajorCompaction(Nemesis):
 
     def disrupt(self):
         self.disrupt_disable_binary_gossip_execute_major_compaction()
+
+
+class ManagerOperationsMonkey(SisyphusMonkey):
```
[the remark this answers is not in the document]
Comment (ShlomiBalalis) [reply sdcm/nemesis.py]: Done

File: configurations/manager/manager_operations.yaml [hunk 2]
```diff
@@ -0,0 +1 @@
+nemesis_class_name: 'ManagerOperationsMonkey'
```
[the remark this answers is not in the document]
Comment (ShlomiBalalis) [reply configurations/manager/manager_operations.yaml]: Done

File: jenkins-pipelines/manager/longevity-large-partition-asymmetric-cluster-ManagerOperationsMonkey-3h.jenkinsfile
Comment (roydahan) [reply jenkins-pipelines/manager/longevity-large-partition-asymmetric-cluster-ManagerOperationsMonkey-3h.jenkinsfile:1]: I didn't notice it's asymmetry.
Ok, leave it.

**Comment by roydahan** [comment]
@Michal-Leszczynski these are good cases, some should already be covered by SCT regression suite, some we already discussed and should be checked.

However, the relevant place to mention these is the test plan for manager 3.2.

Here @ShlomiBalalis covers only one gap for manager regression.

**Comment by ShlomiBalalis** [comment]
@roydahan So do you approve the PR?

**Review by roydahan — APPROVED** [review]

**Review by dkropachev — APPROVED** [review]

**Review by soyacz — APPROVED** [review]

**Comment by roydahan** [comment]
We can backport to manager branch but hold the backports to Scylla branches.
I don't know if we will ever do them.

**Pull request merged**